File: src/main/java/com/esprit/ms/microserviceproject/entities/Annonce.java
package com.esprit.ms.microserviceproject.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import lombok.experimental.FieldDefaults;

import javax.persistence.*;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Annonce {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long idAnnouncement;
    String description;
    String name;
    String localisation;
    double price;
    Boolean etat;
   @Enumerated(EnumType.STRING)
    CategorieAnnonce categorieAnnonce;
    String region;
    int likes;
    int dislikes;
    private  String image;
    //double score;
    int numEvaluations;
    @JsonIgnore
    @OneToMany(mappedBy = "annonce")
    List<Visit> visits;

}

File: src/app/shared/Annonce.ts
export class Annonce{
  idAnnouncement!:number;
  description!:string;
  name!:string;
  localisation!:string ;
  likes!:number;
  dislikes!:number;
  price!:DoubleRange;
  etat!:boolean;
  region!:string;
  image!:string;
  score!:DoubleRange;
  numEvaluations!:number;
  categorieAnnouncement!: CategorieAnnouncement;

  }
  export enum CategorieAnnouncement {
    MaisonAvendre = 'MaisonAvendre',
    MaisonALouer = 'MaisonALouer',
    Villa = 'Villa',
    Appartements = 'Appartements',
    Terrain = 'Terrain'
  }
  

    

